### DISCUSSION TITLE
Pulling frontmatter from md file

### DISCUSSION THREAD

**Discussion by Rahul-krishnakumar** [facebook/docusaurus#5594] [Q&A] [answered]
Hi,

Is there any way to pull the frontmatter from the md files? We would like to pull the title and id of the documents

**Comment by slorber** [comment, marked as answer]
I don't understand what you mean by "pull" here

**Reply by Rahul-krishnakumar** [reply]
I mean to get the frontmatter values as variables we can use. Like maybe an object or something?

**Reply by slorber** [reply]
I'm not sure where you want to use this frontmatter.

Your MDX document contains a `frontMatter` variable in its scope that you can use inside the document and forward to included React components.

This should work:

```md
---
hey: "test"
---

# title

<div>{JSON.stringify(frontMatter)}</div>
```

**Reply by Rahul-krishnakumar** [reply]
So this can only be accessed from inside the document right? There's no way to query it from say another component or hook?

**Reply by slorber** [reply]
@Rahul-krishnakumar I can't really answer your question because you don't expose your use case in detail. Just tell me what you want to do: where do you want to use/display a doc's frontmatter exactly?

**Reply by Rahul-krishnakumar** [reply]
@slorber We want to use the frontmatter for creating an index for searching. We can't use Algolia since it would mean sending data outside our network. It would also help for adding filters to search as well.

**Reply by slorber** [reply]
There are probably local search plugins to take inspiration from. 
As far as I know, most plugins use `postBuild` and read the generated HTML files to create that index. 
It may be possible to provide the data as json in a hook but I don't think we have any infrastructure to do that currently.

The frontmatter is injected inside the Doc theme components rendered by React, so you should be able to swizzle one comp and add any page data you want, including frontmatter serialized in json or some custom head metadatas. 
You can use `@docusaurus/Head` and a wrapper around `@theme/DocItem` for that: https://docusaurus.io/docs/using-themes#for-site-owners
